typespec-tests/src/main/java/com/cadl/armstreamstyleserialization/fluent/models/OutputOnlyModelInner.java

```diff
@@ -0,0 +1,215 @@
+     */
+    private String name;
+
+    /*
+     * The id property.
+     */
+    private String id;
+
+    /*
+     * The properties property.
+     */
+    private OutputOnlyModelProperties innerProperties = new OutputOnlyModelProperties();
+
+    /**
+     * Creates an instance of OutputOnlyModelInner class.
+     */
+    protected OutputOnlyModelInner() {
+    }
+
+    /**
+     * Get the kind property: Discriminator property for OutputOnlyModel.
+     * 
+     * @return the kind value.
+     */
+    public String kind() {
+        return this.kind;
+    }
+
+    /**
+     * Get the name property: The name property.
+     * 
+     * @return the name value.
+     */
+    public String name() {
+        return this.name;
+    }
+
+    /**
+     * Set the name property: The name property.
+     * 
+     * @param name the name value to set.
+     * @return the OutputOnlyModelInner object itself.
+     */
+    OutputOnlyModelInner withName(String name) {
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * Get the id property: The id property.
+     * 
+     * @return the id value.
+     */
+    public String id() {
+        return this.id;
+    }
+
+    /**
+     * Set the id property: The id property.
+     * 
+     * @param id the id value to set.
+     * @return the OutputOnlyModelInner object itself.
+     */
+    OutputOnlyModelInner withId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    /**
+     * Get the innerProperties property: The properties property.
+     * 
+     * @return the innerProperties value.
+     */
+    private OutputOnlyModelProperties innerProperties() {
+        return this.innerProperties;
+    }
+
+    /**
+     * Set the innerProperties property: The properties property.
+     * 
+     * @param innerProperties the innerProperties value to set.
+     * @return the OutputOnlyModelInner object itself.
+     */
+    OutputOnlyModelInner withInnerProperties(OutputOnlyModelProperties innerProperties) {
+        this.innerProperties = innerProperties;
+        return this;
+    }
+
+    /**
+     * Get the title property: The title property.
+     * 
+     * @return the title value.
+     */
+    public String title() {
+        return this.innerProperties() == null ? null : this.innerProperties().title();
+    }
+
+    /**
+     * Validates the instance.
+     * 
+     * @throws IllegalArgumentException thrown if the instance is not valid.
+     */
+    public void validate() {
+        if (name() == null) {
+            throw LOGGER.atError()
+                .log(new IllegalArgumentException("Missing required property name in model OutputOnlyModelInner"));
+        }
+        if (innerProperties() == null) {
+            throw LOGGER.atError()
+                .log(new IllegalArgumentException(
+                    "Missing required property innerProperties in model OutputOnlyModelInner"));
+        } else {
+            innerProperties().validate();
+        }
+    }
```

Comment:
Seems we'll not validate read-only required properties: https://github.com/Azure/autorest.java/blob/6fabb93051072929ea2641bcae63a3276fa7fd50/javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java#L1101

Here `id` is read-only and required, and we don't check it in `validate()`.
So the scenario where we `validate()` locally will only limit to flatten properties in parent class.

Reply:
`validate()` is for sending the request.

If property is `readonly`, it won't be in the request at all.

Reply:
As offline discussed, we'll validate all properties locally.